cmd/export/export.go

```diff
@@ -18,161 +16,49 @@ import (
 )
 
 type ExportOptions struct {
+	configFlags *genericclioptions.ConfigFlags
+
 	// Two GlobalFlags struct fields are needed
 	// 1. cobraGlobalFlags for explicit CLI args parsed by cobra
 	// 2. globalFlags for the args merged with values from the viper config file
 	cobraGlobalFlags *flags.GlobalFlags
 	globalFlags      *flags.GlobalFlags
-	// Two Flags struct fields are needed
-	// 1. cobraFlags for explicit CLI args parsed by cobra
-	// 2. Flags for the args merged with values from the viper config file
-	cobraFlags Flags
-	Flags
 
-	configFlags *genericclioptions.ConfigFlags
-	extras      map[string][]string
-	genericclioptions.IOStreams
-}
-
-type Flags struct {
-	ExportDir string `mapstructure:"export-dir"`
-	Context   string `mapstructure:"context"`
-	Namespace string `mapstructure:"namespace"`
+	rawConfig api.Config
+	exportDir string `mapstructure:"export-dir"`
+	// TODO(djzager): how do we handle viper + k8s cli options?
+	namespace string
```

Comment:
@sseago brought up, and we need to answer...how are we going to handle the flags-file overrides. It's not immediately obvious how to make that work.